flowzone.yml

```diff
@@ -1196,17 +1196,33 @@ jobs:
       branch_tag: ${{ steps.meta.outputs.branch_tag }}
       sha_tag: ${{ steps.meta.outputs.sha_tag }}
       version_tag: ${{ steps.meta.outputs.version_tag }}
+      has_npm_lockfile: ${{ steps.cached.outputs.has_npm_lockfile }}
 
     steps:
       - *downloadSourceArtifact
       - *extractSourceArtifact
 
-      - name: Setup Node.js
+
+      - name: Check if node-setup can be cached
+        id: cached
+        run: |
+          has_npm_lockfile="$([ -e package-lock.json ] || [ -e npm-shrinkwrap.json ] && echo true || echo false)"
+          echo "has_npm_lockfile=${has_npm_lockfile}" >> $GITHUB_OUTPUT
```

Comment:
This should probably be under `npm_check`

Reply:
+1

Reply:
moved